### PR TITLE
Move dynamic bindings (resource-timing) to requests.js

### PR DESCRIPTION
**Pull request by lannka**

### CODE REVIEW

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 1]
```diff
@@ -511,30 +503,6 @@ export class AmpAnalytics extends AMP.BaseElement {
...
-    if (resourceTimingSpec) {
-      // Check if we're done reporting resource timing metrics before binding
-      // before binding the resource timing variable.
-      if (!resourceTimingSpec['done'] &&
-          Date.now() < this.maxResourceTimingReportingTime_) {
-        const binding = 'RESOURCE_TIMING';
-        const analyticsVar = 'resourceTiming';
-        dynamicBindings[binding] =
-            serializeResourceTiming(this.win, resourceTimingSpec);
-        expansionOptions.vars[analyticsVar] = binding;
```
Comment (lannka) [inline extensions/amp-analytics/0.1/amp-analytics.js]: the method has a side effect in this line which mutates the param. 
removed and moved to the mapping to vendors.js.

File: extensions/amp-analytics/0.1/variables.js [hunk 2]
```diff
@@ -144,15 +145,36 @@ export class VariableService {
     this.register_('$IF',
         (value, thenValue, elseValue) => value ? thenValue : elseValue);
     this.register_('$REPLACE', replaceMacro);
+
+    /**
+     * Maximum time (since epoch) to report resource timing metrics.
+     * We stop reporting after 1 minute.
+     * @private @const {number}
+     */
+    this.maxResourceTimingReportingTime_ = Date.now() + (60 * 1000);
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/variables.js]: This doesn't work. It's each analytics instance has single reportingTime. the `getMacroBindings` only gets called before sending request so we can't start the timer with the `getMacroBindings` as well.

File: extensions/amp-analytics/0.1/amp-analytics.js (hunk 1 above)
Comment (zhouyx) [reply extensions/amp-analytics/0.1/amp-analytics.js]: Sounds good.

File: extensions/amp-analytics/0.1/variables.js (hunk 2 above)
Comment (lannka) [reply extensions/amp-analytics/0.1/variables.js]: good point. moved to requests.js to ensure start time is aligned with the start of amp-analytics element

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 3]
```diff
@@ -511,30 +503,6 @@ export class AmpAnalytics extends AMP.BaseElement {
...
-   * @return {!Object<string, (string|!Promise<string>|function(): string)>}
-   * @private
-   */
-  getDynamicVariableBindings_(trigger, expansionOptions) {
-    const dynamicBindings = {};
-    const resourceTimingSpec = trigger['resourceTimingSpec'];
-    if (resourceTimingSpec) {
-      // Check if we're done reporting resource timing metrics before binding
-      // before binding the resource timing variable.
-      if (!resourceTimingSpec['done'] &&
```
[the remark this answers is not in the document]
Comment (lannka) [reply extensions/amp-analytics/0.1/amp-analytics.js]: That's right. it was too arbitrary that sometimes return empty string and sometimes it keeps the RESOURCE_TIMING, we'd better to unify the behavior.

File: extensions/amp-analytics/0.1/resource-timing.js [hunk 4]
```diff
@@ -289,3 +289,18 @@ export function serializeResourceTiming(win, resourceTimingSpec) {
...
+
+/**
+ * @param {!Window} win resource timing spec.
+ * @param {!JsonObject|undefined} spec resource timing spec.
+ * @param {number} startTime start timestamp.
+ * @return {!Promise<string>}
+ */
+export function getResourceTiming(win, spec, startTime) {
+  // Only allow collecting timing within 1s
+  if (spec && (Date.now() < (startTime + 60 * 1000))) {
```
[the remark this answers is not in the document]
Comment (lannka) [reply extensions/amp-analytics/0.1/resource-timing.js]: right, it does not make too much sense to store that time in requests.js

File: extensions/amp-analytics/0.1/test/test-amp-analytics.js [hunk 5]
```diff
@@ -1716,88 +1711,4 @@ describes.realWin('amp-analytics', {
...
-        expect(sendRequestSpy.args[0][0]).to.equal(expectedPing);
-      });
-    };
-
-    it('should evaluate ${resourceTiming} to be empty by default', () => {
-      return runResourceTimingTest(
-          [], newConfig(), 'https://ping.example.com/endpoint?rt=');
-    });
-
-    it('should capture multiple matching resources', () => {
```
[the remark this answers is not in the document]
Comment (lannka) [reply extensions/amp-analytics/0.1/test/test-amp-analytics.js]: there is already a test case in test-resource-timing. see L177

File: extensions/amp-analytics/0.1/test/test-amp-analytics.js [hunk 6]
```diff
@@ -1716,88 +1711,4 @@ describes.realWin('amp-analytics', {
...
-          [entry1, entry2], config,
-          'https://ping.example.com/endpoint?rt=' +
-              'foo_bar-script-100-500-7200~' +
-              'foo_bar-script-700-100-0');
-
-      // 'responseAfter' should be set to a positive number.
-      expect(trigger['resourceTimingSpec']['responseAfter']).to.be.above(0);
-    });
-
-    it('should url encode variables', () => {
```
[the remark this answers is not in the document]
Comment (lannka) [reply extensions/amp-analytics/0.1/test/test-amp-analytics.js]: same here, see L349

File: extensions/amp-analytics/0.1/amp-analytics.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (zhouyx) [reply extensions/amp-analytics/0.1/amp-analytics.js]: SGTM

**Pull request merged**